resources/telemetry/charts/fluent-bit/templates/_pod.tpl

```diff
@@ -100,8 +100,20 @@ containers:
     {{- if .Values.extraVolumeMounts }}
       {{- toYaml .Values.extraVolumeMounts | nindent 6 }}
     {{- end }}
-  {{- if .Values.extraContainers }}
-    {{- toYaml .Values.extraContainers | nindent 2 }}
+  {{- if .Values.exporter.enabled }}
```

Comment:
It would be good to utilize the extraContainers value and add the sidecar via the values file. Mind that the Fluent Bit chart is from the upstream project and any modification to the templates will be additional work in future when updating the chart.